src/Acuminator/Acuminator.Analyzers/StaticAnalysis/EventHandlerModifier/EventHandlerProtectedModifierFix.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System.Composition;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CodeActions;
+using System.Collections.Generic;
+using System.Threading;
+using System.Linq;
+
+namespace Acuminator.Analyzers.StaticAnalysis.EventHandlerModifier
+{
+	[ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+	public class EventHandlerProtectedModifierFix : PXCodeFixProvider
+	{
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			ImmutableArray.Create(
+				Descriptors.PX1077_EventHandlersShouldBeProtectedVirtual.Id,
+				Descriptors.PX1077_EventHandlersShouldNotBeExplicitInterfaceImplementations.Id);
+
+		protected override async Task RegisterCodeFixesForDiagnosticAsync(CodeFixContext context, Diagnostic diagnostic)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			var root = await context.Document
```

Comment:
You should not retrieve the syntax root and run async operation that you can avoid before you have checked diagnostic properties.

Reply:
Done.